utilities/upgrade.py

```
@@ -0,0 +1,180 @@
+    Args:
+        script_path (str): Full path to the upgrade script (e.g., "/path/to/mtv-autodeploy/mtv-upgrade.sh").
+        mtv_version (str): Target MTV version to upgrade to.
+        mtv_source (str): MTV source identifier (e.g., "brew", "released").
+        image_index (str): Optional image index override for the upgrade.
+
+    Raises:
+        ValueError: If required parameters are empty.
+        MtvUpgradeError: If the upgrade script exits with a non-zero status.
+    """
+    if not script_path:
+        raise ValueError("script_path must be provided via --tc=upgrade_script_path:<path>")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)